app/src/main/java/org/mozilla/focus/utils/AndroidBundleSerializer.java

```diff
@@ -0,0 +1,151 @@
+package org.mozilla.focus.utils;
+
+import android.os.Bundle;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class AndroidBundleSerializer {
+
+    private final Map<String, DataTypeHandler> dataTypeHandlers;
+
+    public interface DataTypeHandler {
+        String getName();
```

Comment:
I suggest to use `getSimpleName` since the implementation do use `getSimpleName`

Reply:
I prefer not to leak the implementation detail for this method.